relentless/core.py

```diff
@@ -82,7 +82,7 @@ def __call__(self, x):
 
```

Comment:
I'm tagging a few other doc changes for lines above:
- [ ] Add explicit test in `__init__` that x is increasing before setting the domain. You can use something like:
```py
x = np.atleast_1d(x)
if not np.all(x[1:] > x[:-1]):
    raise ValueError('x must be increasing')
```
You might also want to check that x is only a 1d array using `len(x.shape) == 1`.
- [ ] Update the parameters for `__init__` to document that x can be a float or `array_like`
- [ ] Update the parameters for `__call__` to document that x can be a float or `array_like`. Remove comment about being promoted to an array.